src/intelExt.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * Some simple wrappers to enable HElib to use HEXL.
+ */
+
+#ifdef USE_INTEL_HEXL
+
+#include "intelExt.h"
+
+#include <hexl/hexl.hpp>
+
+#include <unordered_map>
+#include <functional>
+#include <shared_mutex>
+#include <mutex>
+
+namespace intel {
+
+using intel::hexl::NTT;
+
+static std::shared_mutex table_mutex;
+
+struct Key
+{
+  uint64_t degree;
+  uint64_t q;
+
+  bool operator==(const Key& other) const
+  {
+    return this->degree == other.degree && this->q == other.q;
+  }
+};
+
+struct Hash
+{
+  size_t operator()(const Key& key) const
+  {
+    return std::hash<uint64_t>{}(key.degree) ^
```

Comment:
I suspect this may in theory allow some hash collisions. Perhaps 
`key.degree ^ (key.q << MAX_LOG2_DEGREE)` or similar?